src/main/java/Duke.java

```diff
@@ -1,10 +1,89 @@
+    public static void markTaskInListAsDone(String command){
+        int dividerPosition = command.indexOf(" ");
+        int taskNumCompleted = Integer.parseInt(command.substring(dividerPosition).trim());
+
+        if (taskNumCompleted > numOfTasks) {
+            System.out.println("This is not a valid task\nPlease enter a valid task number");
+        }
+
+        System.out.println("Good Job completing your task! I have marked it as done.");
+        tasks[taskNumCompleted - 1].markTaskDone();
+        System.out.println(tasks[taskNumCompleted - 1]);
+    }
+
+    // Determine the type of task added and add to the existing list of tasks
+    public static void addTask (String command){
+        int descriptionPosition = command.indexOf(" ");
+        if (command.contains("todo")){
+            tasks[numOfTasks] = new ToDo(command.substring(descriptionPosition).trim());
+        }
+        else if (command.contains ("deadline")){
+            int byPosition = command.indexOf("/");
+            String description = command.substring(descriptionPosition, byPosition).trim();
+            String by = command.substring(byPosition + 3).trim(); // plus 3 to get rid of the /by
+            tasks[numOfTasks] = new Deadline(description, by);
+        }
+        else if (command.contains ("event")) {
+            int atPosition = command.indexOf("/");
+            String description = command.substring(descriptionPosition, atPosition).trim();
+            String at = command.substring(atPosition + 3).trim(); // plus 3 to get rid of the /by
+            tasks[numOfTasks] = new Event(description, at);
+        }
+        else {
+            tasks[numOfTasks] = new Task(command);
+        }
+        System.out.println("There you go I've added it to the list\n" + tasks[numOfTasks]);
+        numOfTasks ++;
+    }
+
+    // Determine the command sent and execute accordingly
+    public static void executeCommand (String command){
+        if (command.equals("bye")){
+            exitDuke();
+        }
+        else if (command.equals("list")) {
+            printList();
+        }
+        else if (command.contains("done")){
+            markTaskInListAsDone(command);
+        }
+        else {
+            addTask(command);
+        }
+    }
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        // set up scanner
+        Scanner in = new Scanner (System.in);
+        String command;
+
+        // Introduction
+        System.out.println ("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+
+        while (!shouldExit){
+            command = in.nextLine();
+            executeCommand(command);
+        }
     }
```

Comment:
Good job in extracting out the classes to make the code more readable. Keep it up!